sdk/storage/azure-storage-blob/V8_V12.md

```diff
@@ -0,0 +1,301 @@
+
+## Prerequisites
+Java Development Kit (JDK) with version 8 or above
+
+## Converting Core Classes
+Our core synchronous classes have been replaced, as well as new a~~~~synchronous counterparts added.
+
+| Core V8 classes  | Equivalent V12 Class | NEW Asynchronous clients |
+|---:|---:|---:|
+| CloudStorageAccount   | BlobServiceClientBuilder  |  BlobServiceClientBuilder |
+| CloudBlobClient   | BlobServiceClient  |  BlobServiceAsyncClient |
+| CloudBlobContainer |  ContainerClient | ContainerAsyncClient | 
+| CloudBlob   | BlobClient | BlobAsyncClient  |   
+| CloudBlockBlob  |  BlockBlobClient  | BlockBlobAsyncClient  |  
+| CloudAppendBlob  |  AppendBlobClient  | AppendBlobAsyncClient | 
+| CloudPageBlob |  PageBlobClient | PageBlobAsyncClient | 
+
+## Updated Maven dependency
+Dependency for Blob service:
+```xml
+<dependency>
+  <groupId>com.azure</groupId>
+  <artifactId>azure-storage-blob</artifactId>
+  <version>12.0.0</version>
+</dependency>
+```
+
+## Generate SAS token
+SAS token generation has moved from helper classes to the core clients themselves.
+See section [SAS Token](#sastoken) to learn how to use them as a credential.
+Not all fields being set in these samples are necessary.
+At minimum, only an expiry time and permission set are necessary to create a SAS.
+
+In V8, to generate the SAS token, you needed to use `SharedAccessSignatureHelper` to fluently build out all it's options. 
+You also needed to provide `ServiceClient` which includes `SharedKeyCredential` to sign the token with.
+```java
+// specify token properties
+CloudStorageAccount storageAccount = CloudStorageAccount.parse("<connection-string>");
+CloudBlobClient blobClient = storageAccount.createCloudBlobClient();
+String identifier = "identifier";
+
+EnumSet<SharedAccessAccountPermissions> permissions = EnumSet.of(
+    SharedAccessAccountPermissions.READ,
+    SharedAccessAccountPermissions.WRITE,
+    SharedAccessAccountPermissions.CREATE,
+    SharedAccessAccountPermissions.DELETE,
+    SharedAccessAccountPermissions.ADD);
+
+IPRange ipR = new IPRange("0.0.0.0", "255.255.255.255");
+
+Calendar calendar = Calendar.getInstance();
+Date startDate = calendar.getTime();
+calendar.add(Calendar.DAY_OF_YEAR, 1);
+Date expiryDate = calendar.getTime();
+
+SharedAccessHeaders headers = new SharedAccessBlobHeaders();
+headers.setCacheControl("cache");
+headers.setContentDisposition("dispoistion");
+headers.setContentEncoding("encoding");
+headers.setContentLanguage("language");
+headers.setContentType("type");
+
+SharedAccessAccountPolicy policy = new SharedAccessAccountPolicy();
+policy.setPermissions(permissions);
+policy.setSharedAccessStartTime(startDate);
+policy.setSharedAccessExpiryTime(expiryDate);
+policy.setServices(EnumSet.of(SharedAccessAccountService.BLOB));
+
+// build the token
+String sasToken = SharedAccessSignatureHelper.generateSharedAccessSignatureHashForBlobAndFile(
```

Comment:
This helper type is actually marked as reserved for internal use. The customer is supposed to call the generate sas methods on the clients.